runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -1086,6 +1087,11 @@ function M.display(diagnostics, bufnr, client_id, config)
     M.set_underline(diagnostics, bufnr, client_id, nil, underline_opts)
   end
 
+  local severity_sort = resolve_optional_value(config.severity_sort)
```

Comment:
Should we maybe do this upstream as diagnostics come in rather than resorting on each display?